faq-site-client/src/components/global/NavDrawer.vue

```diff
@@ -70,6 +72,21 @@
                 }
             }
         },
+        watch: {
+            $route(to: Route, from: Route) {
+                if (to.fullPath.includes(Routes.TOPIC)) {
+                    this.activeTopicHash = [+(to.params as any).hash]; //Perhaps not use this later on, but doing this thru the store
+                }
+            },
+            activeTopicHash(hash: number[]) {
+                if (hash.length !== 0) {
+                    const topicObj = getTopicFromHash(hash[0], this.topics);
+                    if (!router.currentRoute.fullPath.includes(Routes.TOPIC) || topicObj.hash !== +(router.currentRoute.params as any).hash) {
```

Comment:
router is also accessible by `this.$router`

Reply:
Also `router.currentRoute.params as any` is unnecessary`

Reply:
The `as any` I did because IntelliJ would give some errors if I didn't.